### PR TITLE
Add global theme settings

### PR DESCRIPTION
**Pull request by IKupriyanov-HORIS**

### CODE REVIEW

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt [hunk 1]
```diff
@@ -8,4 +8,6 @@ package jetbrains.letsPlot
 interface FrontendContext {
     fun display(plotSpecRaw: MutableMap<String, Any>)
     fun getInfo() = "Not available"
-}
\ No newline at end of file
+}
+
+expect fun createDefaultFrontendContext(): FrontendContext
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt:13]: Lets have this declaration in a separate source file.

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private val settings: MutableMap<String, String> = mutableMapOf()
+
+    fun set(pair: Pair<String, String>) {
+        settings += pair
+    }
+
+
+    private const val PLOT_THEME = "plot_theme"
+
+    fun setTheme(theme: OptionsMap?) {
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt:29]: Let's handle theme similar to `isolatedFrameContext`:
- property (no get/set methods) holding an OptionsMap? value.
- read LETS_PLOT_PLOT_THEME env var to initialize.

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/LetsPlot.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import jetbrains.letsPlot.frontend.NotebookFrontendContext
+import jetbrains.letsPlot.intern.OptionsMap
+
+object LetsPlot {
+    var frontendContext: FrontendContext = createDefaultFrontendContext()
+
+    @Suppress("MemberVisibilityCanBePrivate")
+    var apiVersion: String = "Unknown"
+
+    fun setTheme(theme: OptionsMap?) {
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/LetsPlot.kt:17]: Lets's make it a property - similar to `frontendContext`.

File: plot-api/src/jsMain/kotlin/jetbrains/letsPlot/Env.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2022. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.letsPlot
+
+internal actual object Env {
+    actual fun get(name: String): String? {
+        TODO("Not yet implemented")
```
Comment (alshan) [inline plot-api/src/jsMain/kotlin/jetbrains/letsPlot/Env.kt:10]: Why not return null?

File: plot-api/src/jsMain/kotlin/jetbrains/letsPlot/defaultFrontendContext.kt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2022. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.letsPlot
+
+actual fun createDefaultFrontendContext(): FrontendContext {
+    TODO("Not yet implemented")
```
Comment (alshan) [inline plot-api/src/jsMain/kotlin/jetbrains/letsPlot/defaultFrontendContext.kt:9]: Return an instance throwing a IllegalStateException - similar to JVM variant.

**Review by alshan — CHANGES_REQUESTED** [review]

File: plot-api/src/jsMain/kotlin/jetbrains/letsPlot/Env.kt (hunk 4 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/jsMain/kotlin/jetbrains/letsPlot/Env.kt:10]: Was thinking about using [localStorage](https://kotlinlang.org/api/latest/jvm/stdlib/kotlinx.browser/local-storage.html) as Env in JS. For now will return `null`.

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt (hunk 1 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt]: Done

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt (hunk 2 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt]: Done

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/LetsPlot.kt (hunk 3 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/LetsPlot.kt]: Done

File: plot-api/src/jsMain/kotlin/jetbrains/letsPlot/defaultFrontendContext.kt (hunk 5 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/jsMain/kotlin/jetbrains/letsPlot/defaultFrontendContext.kt]: Done

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/DefaultFrontendContext.kt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2022. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.letsPlot
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/DefaultFrontendContext.kt:6]: Move to package jetbrains.letsPlot.intern.settings.

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/Env.kt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2022. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.letsPlot
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/Env.kt:6]: Move to package jetbrains.letsPlot.intern.settings.

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt [hunk 8]
```diff
@@ -8,4 +8,20 @@ package jetbrains.letsPlot
...
+}
+
+internal fun undefinedFrontendContext() = object : FrontendContext {
+    override fun display(plotSpecRaw: MutableMap<String, Any>) {
+        throw IllegalStateException(
+            """
+                            
+                            The frontend context is not defined.
+                            To define the frontend context please select one of the following options:
+                            a. Add "lets-plot-batik-<version>.jar" to your classpath.
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt:20]: Suggestions a) b) make no sense for JS target.

**Review by alshan — CHANGES_REQUESTED** [review]

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/Env.kt (hunk 7 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/Env.kt]: Done

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt (hunk 8 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/FrontendContext.kt]: Done

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/DefaultFrontendContext.kt (hunk 6 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/DefaultFrontendContext.kt]: Done

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt [hunk 9]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2022. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.letsPlot
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt:6]: Let's move GlobalSettings to the "jetbrains.letsPlot.intern.settings" package.

**Review by alshan — CHANGES_REQUESTED** [review]

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt (hunk 9 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/GlobalSettings.kt]: Done

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/intern/settings/GlobalSettings.kt [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+import jetbrains.datalore.base.json.JsonSupport.parseJson
+import jetbrains.datalore.plot.config.Option.Plot.THEME
+import jetbrains.letsPlot.intern.OptionsMap
+import jetbrains.letsPlot.intern.filterNonNullValues
+
+
+// Environment variables
+const val ENV_HTML_ISOLATED_FRAME = "LETS_PLOT_HTML_ISOLATED_FRAME"
+const val ENV_LETS_PLOT_PLOT_THEME = "LETS_PLOT_PLOT_THEME"
```
Comment (alshan) [inline plot-api/src/commonMain/kotlin/jetbrains/letsPlot/intern/settings/GlobalSettings.kt:16]: rename to 'ENV_PLOT_THEME'

**Review by alshan — CHANGES_REQUESTED** [review]

File: plot-api/src/commonMain/kotlin/jetbrains/letsPlot/intern/settings/GlobalSettings.kt (hunk 10 above)
Comment (IKupriyanov-HORIS) [reply plot-api/src/commonMain/kotlin/jetbrains/letsPlot/intern/settings/GlobalSettings.kt]: Done

**Pull request merged**